src/types/todo.types.ts

```diff
@@ -0,0 +1,85 @@
+/**
+ * @description 타입 정의, 유틸 함수
+ */
+import { TodoListAppStyles } from '../components/todo-list-styles.components';
+
+export type SelectBtnType = 'allItems' | 'activeItems' | 'completedItems' | 'clearCompleted';
+
+export interface TodoListItem {
+  id: string;
+  label: string;
+  isChecked: boolean;
+  createDt: number;
+}
+
+export interface TodoListDefaultLabel {
+  itemCnt: string;
+  allItems: string;
+  activeItems: string;
+  completedItems: string;
+  clearCompleted: string;
+  noItems: string;
+}
+
+export interface TodoListAppOptions {
+  el: HTMLDivElement;
+  placeholder?: string;
+  styles?: TodoListAppStyles;
+  items?: TodoListItem[];
+  useDnd?: boolean;
+  labels?: TodoListDefaultLabel;
+}
+
+export interface TodoListAppLayouts {
+  root: string;
+  ul: string;
+  li: string;
+  input: string;
+  label: string;
+  buttonWrapper: string;
+  buttonPack: string;
+  noItems: string;
+  checkbox: string;
+  count: string;
+  filter: string;
+  clear: string;
+}
+
+/**
+ * 유틸 함수 모음
+ * 싱글톤 클래스(인스턴스x)에 대응하는 목적으로 네이밍 Utils 유지
+ */
+export const Utils = {
+  replaceToken: (template: string, cnt: number) => {
+    return template.replace('#{}', cnt.toString());
+  },
+  moveItem: (arr: TodoListItem[], fromIndex: number, toIndex: number) => {
+    const newArr = [...arr];
+    const [item] = newArr.splice(fromIndex, 1);
+    if (!item) {
+      return arr;
+    }
+    newArr.splice(toIndex, 0, item);
```

Comment:
투두 아이템을 아래 방향으로 dnd 했을때 원래 삽입되어야하는 위치보다 한 칸 뒤로 밀리고 있는데,
아래 방향으로 이동 시, 한 번 제거한 뒤 타겟 인덱스가 한 칸 당겨지므로 보정 로직이 필요할것 같습니다

```suggestion
    const normalizedToIndex = toIndex > fromIndex ? toIndex - 1 : toIndex;
    newArr.splice(normalizedToIndex, 0, item);
```